webviz_subsurface/plugins/_structural_uncertainty/controllers/map_controller.py

```diff
@@ -484,12 +488,18 @@ def check_if_update_needed(
             "map-color-ranges" in ctx["prop_id"]
             and color_range_settings[map_id]["update"]
         )
+        initial_loading = True
+        if current_map:
+            for layer in current_map:
+                if layer.get("id") == map_id:
+                    initial_loading = False
```

Comment:
```suggestion
initial_loading = not current_map or all(layer.get("id") != map_id for layer in current_map))
```
Not sure if this is any better (or correct) - but can at least throw it in as a possible suggestion.